api/xds_protocol.rst

```diff
@@ -36,43 +36,149 @@ occurs.
+   -  SotW: ListenerDiscoveryService.StreamListeners
+   -  Incremental: ListenerDiscoveryService.DeltaListeners
+-  RouteConfiguration: Route Discovery Service (RDS)
+   -  SotW: RouteDiscoveryService.StreamRoutes
+   -  Incremental: RouteDiscoveryService.DeltaRoutes
+-  VirtualHost: Virtual Host Discovery Service (VHDS)
+   -  SotW: N/A
+   -  Incremental: VirtualHostDiscoveryService.DeltaVirtualHosts
+-  Cluster: Cluster Discovery Service (CDS)
+   -  SotW: ClusterDiscoveryService.StreamClusters
+   -  Incremental: ClusterDiscoveryService.DeltaClusters
+-  ClusterLoadAssignment: Endpoint Discovery Service (EDS)
+   -  SotW: EndpointDiscoveryService.StreamEndpoints
+   -  Incremental: EndpointDiscoveryService.DeltaEndpoints
+-  Secret: Secret Discovery Service (SDS)
+   -  SotW: SecretDiscoveryService.StreamSecrets
+   -  Incremental: SecretDiscoveryService.DeltaSecrets
+-  Runtime: Runtime Discovery Service (RTDS)
+   -  SotW: RuntimeDiscoveryService.StreamRuntime
+   -  Incremental: RuntimeDiscoveryService.DeltaRuntime
+
+In the aggregated API variants, all resource types are multiplexed on a single gRPC stream, where each resource type is
+treated as a separate logical stream within the aggregated stream. In effect, it simply combines all of the above separate
+APIs into a single stream by treating requests and responses for each resource type as a separate sub-stream on the single
+aggregated stream. The RPC service and methods for the aggregated API variants are:
+
+-  SotW: AggregatedDiscoveryService.StreamAggregatedResources
+-  Incremental: AggregatedDiscoveryService.DeltaAggregatedResources
+
+For all of the SotW methods, the request type is :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` and the response
+type is :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`.
+
+For all of the incremental methods, the request type is :ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` and
+the response type is :ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`.
+
+Configuring Which Variant to Use
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the xDS API, the :ref:`ConfigSource <envoy_api_msg_core.ConfigSource>` message indicates how to obtain resources of a
+particular type. If the *ConfigSource* contains a gRPC :ref:`ApiConfigSource <envoy_api_msg_core.ApiConfigSource>`, it points
+to an upstream cluster for the management server; this will initiate an independent bidirectional gRPC stream for each xDS
+resource type, potentially to distinct management servers. If the *ConfigSource* contains a :ref:`AggregatedConfigSource
+<envoy_api_msg_core.AggregatedConfigSource>`, it tells the client to use :ref:`ADS <xds_protocol_ads>`.
+
+Currently, the client is expected to be given some local configuration that tells it how to obtain the `Listener` and
+`Cluster` resources. `Listener` resources may include a *ConfigSource* that indicates how the `RouteConfiguration` resources
```

Comment:
Nit: prefer that we are consistent in how we refer to API types. The three options I think that are valid are:
```
`<name`
*<name>*
:ref:`<name> <reference>`
```
there is a mix of these in this section.

Reply:
Done.